### PR TITLE
Update the `codelab_rebuild` script for `adaptive_app`

### PR DESCRIPTION
**Pull request by domesticmouse**
This PR also includes the start of some documentation for the `codelab_rebuild` tooling.

PTAL @miquelbeltran for sanity checking. Thanks!

## Pre-launch Checklist

- [x] I read the [Effective Dart: Style] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-devrel channel on [Discord].

[Effective Dart: Style]: https://dart.dev/guides/language/effective-dart/style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: adaptive_app/step_04/pubspec.yaml [hunk 1]
```diff
@@ -36,12 +36,12 @@ dependencies:
   # The following adds the Cupertino Icons font to your application.
   # Use with the CupertinoIcons class for iOS style icons.
   cupertino_icons: ^1.0.2
-  flex_color_scheme: ^6.1.1
```
Comment (domesticmouse) [inline adaptive_app/step_04/pubspec.yaml:39]: @miquelbeltran Yes, I know you tidied up the ordering of the `pubspec.yaml`. Unfortunately `flutter pub add`
ignores ordering if the packages are unordered. And by default, they are.

I'm hoping I can rectify this situation once https://github.com/flutter/flutter/pull/113873 ships in the next stable

File: tooling/codelab_rebuild/README.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
+# Codelab Rebuild tooling
```
Comment (domesticmouse) [inline tooling/codelab_rebuild/README.md:1]: @miquelbeltran This is meant to explain the tooling I've built for rebuilding the codelabs at each version update.

PTAL and point out things that don't make sense, could be improved, etc. Thanks!

File: tooling/codelab_rebuild/README.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+## Usage
+
+Running the tool requires a `codelab_rebuild.yaml` file. There are examples
+of these configuration files in the codelabs in this repository. To run the
+tool on the `adaptive_app` codelab, run the following commands:
+
+```console
+$ cd ../..    # go back to the root of the flutter/codelabs checkout
+$ codelab_rebuild adaptive_codelab/codelab_rebuild.yaml
```
Comment (miquelbeltran) [inline tooling/codelab_rebuild/README.md:24]:
```suggestion
$ codelab_rebuild adaptive_app/codelab_rebuild.yaml
```

File: tooling/codelab_rebuild/README.md (hunk 2 above)
Comment (miquelbeltran) [reply tooling/codelab_rebuild/README.md:1]: Just found a small typo on the adaptive_app path and that's it.

I wasn't sure what was the tool exactly for until I opened the `codelab_rebuild.yaml` and wow, it does generate the whole codelab code with all the code changes, with "rebuild" I thought it was compiling the code, not actually generating the codelab code from scratch.

What's your approach for creating the `codelab_rebuild.yaml` file? Do you do a git diff and then copy/paste the patch changes into it? I think it is probably worth explaining what is your workflow with it.

Comment (domesticmouse) [reply tooling/codelab_rebuild/README.md:1]: > What's your approach for creating the `codelab_rebuild.yaml` file? Do you do a git diff and then copy/paste the patch changes into it? I think it is probably worth explaining what is your workflow with it.

Good point, I'll write up a walk through of my codelab development process with the rebuild tooling.

**Comment by domesticmouse** [comment]
@miquelbeltran looks like your m@beltran account isn't CLA'd?

**Comment by domesticmouse** [comment]
> @miquelbeltran looks like your m@beltran account isn't CLA'd?

Actually no, this account is clearing CLA checks on other PRs. I think CLAbot is just confused.

**Comment by RedBrogdon** [comment]
CLA Bot sometimes gets confused when a PR includes commits from multiple authors. Since we're sure in this case that all the code is coming from folks that have signed the agreement, you're clear to ignore the bot.

You may need an LGTM from someone other than Miguel, though, since he's not in the Flutter GitHub org?

**Comment by domesticmouse** [comment]
> You may need an LGTM from someone other than Miguel, though, since he's not in the Flutter GitHub org?

That'd be why you are listed as a PR reviewer @RedBrogdon  =)

**Comment by miquelbeltran** [comment]
> > @miquelbeltran looks like your m@beltran account isn't CLA'd?
> 
> Actually no, this account is clearing CLA checks on other PRs. I think CLAbot is just confused.

Arg, probably because I submitted my change proposal using GitHub's review web interface, so it used my actual email address rather than my "anonymous" GitHub email. I will submit my email to the CLA as well.

**Comment by domesticmouse** [comment]
PTAL @RedBrogdon, this PR is ready to land

File: tooling/codelab_rebuild/README.md (hunk 2 above)
Comment (domesticmouse) [reply tooling/codelab_rebuild/README.md:1]: I'll build out this development process in a separate document and get your review @miquelbeltran

**Review by RedBrogdon — APPROVED** [review]
The README is my favorite part. 😄

**Comment by domesticmouse** [comment]
> The README is my favorite part. 😄

I'm getting there on the documentation front. Ironic, no? =)

**Pull request merged**